packages/bff/src/services/agreementService.ts

```diff
@@ -103,6 +115,104 @@ export function agreementServiceBuilder(clients: PagoPAInteropBeClients) {
 
       return enrichAgreement(agreement, clients, ctx);
     },
+
+    async addAgreementConsumerDocument(
+      agreementId: string,
+      doc: bffApi.addAgreementConsumerDocument_Body,
+      { headers, logger }: WithLogger<BffAppContext>
+    ): Promise<Buffer> {
+      const documentPath = `${config.consumerDocumentsPath}/${agreementId}`;
+      const documentContent = Buffer.from(await doc.doc.arrayBuffer());
+      const documentId = randomUUID();
+
+      await fileManager.storeBytes(
+        config.consumerDocumentsContainer,
+        documentPath,
+        documentId,
+        doc.doc.name,
+        documentContent,
+        logger
+      );
+
+      const seed: agreementApi.DocumentSeed = {
+        id: documentId,
+        prettyName: doc.prettyName,
+        name: doc.doc.name,
+        contentType: doc.doc.type,
+        path: documentPath,
+      };
+
+      await agreementProcessClient.addAgreementConsumerDocument(seed, {
+        params: { agreementId },
+        headers,
+      });
+
+      return documentContent;
+    },
+
+    async getAgreementConsumerDocument(
+      agreementId: string,
+      documentId: string,
+      { headers, logger }: WithLogger<BffAppContext>
+    ): Promise<Buffer> {
+      const documentSeed =
```

Comment:
Minor, but this is not a "seed" since you are not using it to create something new
```suggestion
      const document =
```